algorithms/linfa-pls/benches/pls.rs

```diff
@@ -0,0 +1,93 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use linfa::traits::Fit;
+use linfa::Dataset;
+use linfa_datasets::generate::make_dataset;
+use linfa_pls::Algorithm;
+use linfa_pls::{PlsCanonical, PlsCca, PlsRegression};
+use statrs::distribution::{DiscreteUniform, Laplace};
+
+#[allow(unused_must_use)]
+fn pls_regression(dataset: &Dataset<f64, f64>, alg: Algorithm) {
+    let model = PlsRegression::params(3)
+        .scale(true)
+        .max_iterations(200)
+        .algorithm(alg);
+    model.fit(&dataset);
+}
+
+#[allow(unused_must_use)]
+fn pls_canonical(dataset: &Dataset<f64, f64>, alg: Algorithm) {
+    let model = PlsCanonical::params(3)
+        .scale(true)
+        .max_iterations(200)
+        .algorithm(alg);
+    model.fit(&dataset);
+}
+#[allow(unused_must_use)]
+fn pls_cca(dataset: &Dataset<f64, f64>, alg: Algorithm) {
+    let model = PlsCca::params(3)
+        .scale(true)
+        .max_iterations(200)
+        .algorithm(alg);
+    model.fit(&dataset);
+}
+
+fn bench(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Linfa_pls");
+    for (alg, name) in [(Algorithm::Nipals, "Nipals"), (Algorithm::Svd, "Svd")] {
+        let sizes: [usize; 3] = [1_000, 10_000, 100_000];
+        let num_feats: [usize; 2] = [5, 10];
+
+        let feat_distr = Laplace::new(0.5, 5.).unwrap();
+        let target_distr = DiscreteUniform::new(0, 5).unwrap();
+
+        let mut pls_regression_id = "Regression-".to_string();
+        pls_regression_id.push_str(name);
+        let mut pls_canonical_id = "Canonical-".to_string();
+        pls_canonical_id.push_str(name);
+        let mut pls_cca_id = "Cca-".to_string();
+        pls_cca_id.push_str(name);
+
+        for size in sizes {
+            for num_feat in num_feats {
```

Comment:
Now that I think about it, we don't need to permute all sample sizes and feature counts. We only really need (1000, 5), (10k, 5), (100k, 5), and (100k, 10). In other words, high feature counts are only really interesting when sample count is high. That will reduce the total number of benchmarks by 12, which is alot.

Reply:
added it here.